plugins/wysiwygarea/plugin.js

```diff
@@ -297,7 +297,7 @@
 				var elements = doc.getElementsByTag( tagName );
 				if ( lockRetain ) {
 					if ( elements.count() == 1 && !elements.getItem( 0 ).getCustomData( 'retain' ) &&
-						!elements.getItem( 0 ).hasAttribute( 'data-cke-temp' ) ) {
+						CKEDITOR.tools.isEmpty( elements.getItem( 0 ).getAttributes() ) ) {
```

Comment:
Entire function `removeSuperfluousElement` was design to remove additional `div` which appear in Edge, when all text is removed from editor. This div never had any attribute. Unfortunately author hadn't predict that not only those extra `divs` might appear, but also widgets might be construct based on `div`. And in case of `undo` command entire widget might be redraw so we lost `retain` customData attached to `div`.
This change detects more carefully only those extra divs from Edge, so it becomes more specific. Now divs connected to widgets won't be removed.

Reply:
Makes sense 👍